sdk/src/main/java/io/dapr/client/DaprClientGrpc.java

```diff
@@ -1206,4 +1213,43 @@ public void onCompleted() {
       }
     };
   }
-}
+
+  @Override
+  public Mono<DaprMetadata> getMetadata() {
+    DaprProtos.GetMetadataRequest metadataRequest = DaprProtos.GetMetadataRequest.newBuilder().build();
+    return Mono.deferContextual(
+        context -> this.<DaprProtos.GetMetadataResponse>createMono(
+            it -> intercept(context, asyncStub).getMetadata(metadataRequest, it)))
+        .map(
+            it -> {
+              try {
+                return buildDaprMetadata(it);
+              } catch (IOException ex) {
+                throw DaprException.propagate(ex);
+              }
+            });
+  }
+
+  private DaprMetadata buildDaprMetadata(
+      DaprProtos.GetMetadataResponse response) throws IOException {
+    List<RegisteredComponents> registeredComponentsList = response.getRegisteredComponentsList();
+
+    List<ComponentMetadata> components = new ArrayList<>();
+    for (RegisteredComponents rc : registeredComponentsList) {
+      components.add(new ComponentMetadata(rc.getName(), rc.getType(), rc.getVersion()));
+    }
+
+    List<PubsubSubscription> subscriptionsList = response.getSubscriptionsList();
+    List<SubscriptionMetadata> subscriptions = new ArrayList<>();
+    for (PubsubSubscription s : subscriptionsList) {
+      List<PubsubSubscriptionRule> rulesList = s.getRules().getRulesList();
+      List<RuleMetadata> rules = new ArrayList<>();
+      for (PubsubSubscriptionRule r : rulesList) {
+        rules.add(new RuleMetadata(r.getPath()));
+      }
+      subscriptions.add(new SubscriptionMetadata(s.getTopic(), s.getPubsubName(), s.getDeadLetterTopic(), rules));
+    }
+
+    return new DaprMetadata(response.getId(), response.getRuntimeVersion(), components, subscriptions);
```

Comment:
Is the intent to start with the pubsub and components first? [According to the docs](https://docs.dapr.io/reference/api/metadata_api/), it looks like there is more info we can get like: `httpendpoints`, `enabled features`, `app conn details`, `attributes`.

Reply:
yeah.. we can add this, but I was wondering if we can do it incrementally, as a second PR will be much more scoped and easier from someone in the community to take it as a good first issue.

Reply:
That is fine to have a follow up issue to add the other things not in this initial PR and put the 'good first issue' label on it 👍🏻

Reply:
[I opened an issue here](https://github.com/dapr/java-sdk/issues/1055)